Diagnostic/Utils/XmlUtil.py

```diff
@@ -33,6 +33,14 @@ def addElement(xml,path,el,selector=[]):
             continue
         element.append(el)
 
+
+def addElementToFirstMatchingPathOnly(xml, path, el, selector=[]):
+    element = xml.find(path)
+    if selector and element.get(selector[0]) != selector[1]:
```

Comment:
I would think "MatchingPath" in combination with a non-empty selector would imply "the element which matches the path and which satisfies the selector"; that is, if there's a non-empty selector, keep looking through the document until you find a path match that also satisfies the selector.